lib/features/session/presentation/session_controller.dart

```diff
@@ -85,39 +89,41 @@ class SessionController extends ReloadableController {
     consumeState(_getStoredSessionInteractor.execute());
   }
 
   void _getSession() async {
     consumeState(_getSessionInteractor.execute());
   }
 
-  void _handleSessionFailure(Failure failure) {
-    log('SessionController::_handleSessionFailure(): $failure');
-    if (failure is GetSessionFailure) {
-      final sessionException = failure.exception;
-      var errorMessage = '';
-      if (_checkUrlError(sessionException) && currentContext != null) {
-        errorMessage = AppLocalizations.of(currentContext!).wrongUrlMessage;
-      } else if (sessionException is BadCredentialsException && currentContext != null) {
-        errorMessage = AppLocalizations.of(currentContext!).badCredentials;
-      } else if (sessionException is ConnectionError && currentContext != null) {
-        errorMessage = AppLocalizations.of(currentContext!).connectionError;
-      } else if (sessionException is UnknownError && currentContext != null) {
-        if (sessionException.message != null && sessionException.code != null) {
-          errorMessage = '[${sessionException.code}] ${sessionException.message}';
-        } else if (sessionException.message != null) {
-          errorMessage = sessionException.message!;
-        } else {
-          errorMessage = AppLocalizations.of(currentContext!).unknownError;
-        }
+  void _handleSessionFailure(FeatureFailure failure) {
+    final sessionException = failure.exception;
+    var errorMessage = '';
+    if (_checkUrlError(sessionException) && currentContext != null) {
+      errorMessage = AppLocalizations.of(currentContext!).wrongUrlMessage;
+    } else if (sessionException is BadCredentialsException && currentContext != null) {
+      errorMessage = AppLocalizations.of(currentContext!).badCredentials;
+    } else if (sessionException is ConnectionError && currentContext != null) {
+      errorMessage = AppLocalizations.of(currentContext!).connectionError;
+    } else if (sessionException is UnknownError && currentContext != null) {
+      if (sessionException.message != null && sessionException.code != null) {
+        errorMessage = '[${sessionException.code}] ${sessionException.message}';
+      } else if (sessionException.message != null) {
+        errorMessage = sessionException.message!;
+      } else {
+        errorMessage = AppLocalizations.of(currentContext!).unknownError;
       }
+    }
 
-      logError('SessionController::_handleSessionFailure():errorMessage: $errorMessage');
-      if (errorMessage.isNotEmpty && currentOverlayContext != null) {
-        _appToast.showToastErrorMessage(currentOverlayContext!, errorMessage);
-      }
+    logError('SessionController::_handleSessionFailure():errorMessage: $errorMessage');
+    if (errorMessage.isNotEmpty && currentOverlayContext != null) {
+      _appToast.showToastErrorMessage(currentOverlayContext!, errorMessage);
+    }
+
+    if (PlatformInfo.isMobile) {
+      _handleGetStoredSession();
```

Comment:
Recursive point

Reply:
why not handle it directly in `interceptor`?

Reply:
> why not handle it directly in `interceptor`?

Why interceptor? IMO it is interactor?

Reply:
I mean, `presentation` no need to care it is come from `server` or `local`. Only give `presentation` the result. Keep the logic of get in somewhere (maybe the interceptor, when error in `server`, get from `local`....)

Reply:
Before, we only wanted to call it when we lost network connection and on mobile so we handled it at the presentation layer. Furthermore, we always want to get the latest session from the server because it can change at any time. But it leads to many error cases like this issue, so we will handle it directly in the interactor